pkg/scheduler/scheduling_application.go

```diff
@@ -28,14 +28,22 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/apache/incubator-yunikorn-core/pkg/cache"
+	"github.com/apache/incubator-yunikorn-core/pkg/common"
 	"github.com/apache/incubator-yunikorn-core/pkg/common/resources"
 	"github.com/apache/incubator-yunikorn-core/pkg/events"
 	"github.com/apache/incubator-yunikorn-core/pkg/log"
 	"github.com/apache/incubator-yunikorn-core/pkg/plugins"
 	"github.com/apache/incubator-yunikorn-scheduler-interface/lib/go/si"
 )
 
+const DisableReservationVarName = "DISABLE_RESERVATION"
+
 var reservationDelay = 2 * time.Second
+var reservationDisabled = isReservationDisabled()
+
+func isReservationDisabled() bool {
+	return common.GetBoolEnvVar(DisableReservationVarName, false)
+}
```

Comment:
This will lookup for the env var all the time in each scheduling cycle, this might be too expensive.
To update the env vars, we require a restart (i.e update the ENV VARs defined in the deployment yaml file and restart the scheduler). In this case, I suggest we parse the env vars into a common struct `schedulerOptions`, and pass that to `ClusterSchedulingContext`. So the scheduling code can directly get the options from the context.